### ISSUE TITLE
communication from server client is somehow broken

### ISSUE THREAD

**Issue by MarkoGrbec** [godotengine/godot#97150]
### Tested versions

reproducible in 4.4.dev2
not reproducible in 4.3

### System information

Linux manjaro

### Issue description

if I start with self send it works normally as it makes local host.
if I make first server than other instance client with the options in debugging (more instances (customize run instances))

if I do Register it doesn't want to go through the secure connection.
says:
```
E 0:03:42:0268   process_simplify_path: The rpc node checksum failed. Make sure to have the same methods on both nodes. Node path: Node3D2/NetworkNode
  <C++ Source>   modules/multiplayer/scene_cache_interface.cpp:121 @ process_simplify_path()
E 0:03:42:0268   _process_rpc: RPC - 'Node3D(NetworkNode.gd)::cmd_get_secret_key_from_beginning': Method expected 0 arguments, but called with 1
  <C++ Source>   modules/multiplayer/scene_rpc_interface.cpp:290 @ _process_rpc()
```
as in the remote it's not a `NetworkNode`
but it's actually `@Node3D@132`

that's only my assumption that that's the reason

I've been digging where could be the reason for the error for not able to register about 2 hours
and I'll need about 2 days to make MRP as code is complicated and I forgot most about basic networking to make MRP so I'll need to relearn it.

### Steps to reproduce

1. make customize run instances (at least 2)
2. 1. instance start server
3. 2. instance start client
4. on client (2. instance) register and error should be in debugger

### Minimal reproduction project (MRP)

https://drive.proton.me/urls/6Q6Y6DJXV0#jb53Dp5yzLcZ

sorry it didn't let me place folder in here.

will provide minimal in 2 days for now it's a big project

**Comment by AThousandShips** [comment]
Sounds from the error like you declared the methods differently on the server and the client, but will wait with investigating until you've uploaded a minimal project

**Comment by MarkoGrbec** [comment]
all I can say today before on 4.3 stable it worked well I didn't touched that part of the code

as I wanted to do it on 4.4 dev2 I got the error.

**Comment by akien-mga** [comment]
Duplicate of #96889, which will be fixed in 4.4.dev3.

**Comment by MarkoGrbec** [comment]
Fixed in 4.4dev3.
